test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -499,6 +523,40 @@ public async Task DisposeOnDisconnect()
         await disposed.WaitAsync(this.TimeoutToken);
     }
 
+    [Fact]
+    public async Task SubTypeMarshalable_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableWithSubTypes();
+        IMarshalableWithSubTypes? proxy = await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+    }
+
+    [Fact]
+    public async Task SubTypeMarshalable_SubType1_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableSubType1();
+        IMarshalableSubType1? proxy = (IMarshalableSubType1?)await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+        Assert.Equal(2, await proxy!.GetPlusOneAsync(1));
+    }
+
+    [Fact]
+    public async Task SubTypeMarshalable_SubType2_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableSubType2();
+        IMarshalableSubType2? proxy = (IMarshalableSubType2?)await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+        Assert.Equal(3, await proxy!.GetPlusTwoAsync(1));
+    }
+
+    [Fact]
+    public async Task SubTypeMarshalable_UnknownSubType_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableUnknownSubType();
+        IMarshalableWithSubTypes? proxy = await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+    }
+
```

Comment:
Where is your null object test? We don't want to throw NRE when you call `obj.GetType()`. 😬

Reply:
What does this comment refer to?

Reply:
I'm looking for a test where you pass `null` in as your marshalable object. Such that if we have any code that calls `GetType` on it, it would thrown NRE if we didn't have an appropriate null check.